lib/modules/manager/custom/jsonata/readme.md

```diff
@@ -0,0 +1,182 @@
+- `extractVersion`, or use an `extractVersionTemplate` config field
+- `currentDigest`
+- `registryUrl`, or a use `registryUrlTemplate` config field. If it's a valid URL, it will be converted to the `registryUrls` field as a single-length array
+- `indentation`. It must be either empty, or whitespace only (otherwise `indentation` will be reset to an empty string)
+
+### Usage
+
+To configure it, use the following syntax:
+
+```javascript
+{
+  "customManagers": [
+    {
+      "customType": "jsonata",
+      "fileFormat": "json",
+      "fileMatch": ["<file match pattern>"],
+      "matchStrings": ['<query>'],
+      ...
+    }
+  ]
+}
+```
+
+Where `<query>` is a [JSONata](https://docs.jsonata.org/overview.html) query that transform the contents into a JSON object with the following schema:
+
+To be effective with the JSONata manager, you should understand jsonata queries. But enough examples may compensate for lack of experience.
+
+#### Example queries
+
+Below are some example queries for the generic JSON manager.
+You can also use the [JSONata test website](https://try.jsonata.org) to experiment with queries.
+
+```json title="Dependencies spread in different nodes, and we want to limit the extraction to a particular node"
+{
+  "production": [
+    {
+      "version": "1.2.3",
+      "package": "foo"
+    }
+  ],
+  "development": [
+    {
+      "version": "4.5.6",
+      "package": "bar"
+    }
+  ]
+}
+```
+
+Query:
+
+```
+production.{ "depName": package, "currentValue": version }
+```
+
+```json title="Dependencies spread in different nodes, and we want to extract all of them as if they were in the same node"
+{
+  "production": [
+    {
+      "version": "1.2.3",
+      "package": "foo"
+    }
+  ],
+  "development": [
+    {
+      "version": "4.5.6",
+      "package": "bar"
+    }
+  ]
+}
+```
+
+Query:
+
+```
+*.{ "depName": package, "currentValue": version }
+```
+
+```json title="The dependency name is in a JSON node name and the version is in a child leaf to that node"
+{
+  "foo": {
+    "version": "1.2.3"
+  },
+  "bar": {
+    "version": "4.5.6"
+  }
+}
+```
+
+Query:
+
+```
+$each(function($v, $n) { { "depName": $n, "currentValue": $v.version } })
+```
+
+```json title="The name of the dependency and the version are both value nodes of the same parent node"
+{
+  "packages": [
+    {
+      "version": "1.2.3",
+      "package": "foo"
+    },
+    {
+      "version": "4.5.6",
+      "package": "bar"
+    }
+  ]
+}
+```
+
+Query:
+
+```
+packages.{ "depName": package, "currentValue": version }
+```
+
+```json title="The name of the dependency and the version are in the same string"
+{
+  "packages": ["foo@1.2.3", "bar@4.5.6"]
+}
+```
+
+Query:
+
+```
+$map($map(packages, function ($v) { $split($v, "@") }), function ($v) { { "depName": $v[0], "currentVersion": $v[1] } })
+```
+
+```json title="JSONata manager config to extract deps from package.json file in the renovate repository"
```

Comment:
```suggestion
```json title="JSONata manager config to extract deps from a `package.json` file in the Renovate repository"
```